services/aster-nix/src/fs/file_table.rs

```diff
@@ -76,21 +82,23 @@ impl FileTable {
         };
 
         let min_free_fd = get_min_free_fd();
+        let entry = FileTableEntry::new(file, is_cloexec);
         self.table.put_at(min_free_fd, entry);
         Ok(min_free_fd as FileDescripter)
     }
 
-    pub fn insert(&mut self, item: Arc<dyn FileLike>) -> FileDescripter {
-        let entry = FileTableEntry::new(item, false);
+    pub fn insert(&mut self, item: Arc<dyn FileLike>, is_cloexec: bool) -> FileDescripter {
```

Comment:
I think we have discussed this point before: having boolean arguments could be confusing to a reader. Imagine you are guy who is familiar with UNIX or Linux, but not the specific file table implementation in Asterinas. You are given the code below.

```rust
file_table.insert(file, false);
```

How do you interpret the `false` value? You do not have a clue regarding the meaning of `false`. And you have to look up the documentation for its meaning. What if you define a file table flag type?

```rust
bitflags! {
    pub struct FileTableEntryFlags: u8 {
        const CLOEXEC = 1 << 0;
    }
}
```

Then, the user code would look like below.

```rust
file_table.insert(file, FileTableEntryFlags::empty());
```

The resulting code is absolutely clear. Another benefit of introducing this flag is that this solution is more extensible. We could add more flags to file table entries in the future.

If writing out the empty flag is kind of tedious for the writer and a distraction for the reader, so I think it is probably a good idea to have two version of the same API.

```rust
impl FileTable {
    pub fn insert(&mut self, item: Arc<dyn FileLike>) -> FileDescriptor { ... }

    pub fn insert_with_flags(&mut self, item: Arc<dyn FileLike>, flags: FileTableEntryFlags) -> FileDescriptor { ... }

    pub fn insert_at(
        &mut self,
        fd: FileDescripter,
        item: Arc<dyn FileLike>,
    ) -> Option<Arc<dyn FileLike>> { ... }
}
```

Reply:
I choose to use a `boolen` to handle this case because there is just one fd flag ( CLOEXEC). So it seems that using a boolean value is simple enough.

It must be admitted that using 'biflag' does improve readability over a boolean value. Although a bool value could solve the problem, I accept this suggestion for the sake of more readable code."

Reply:
The `insert_at` with fd flags is  used in `dup3` syscall